src/nsmgr/internal/pkg/manager/manager.go

```diff
@@ -0,0 +1,151 @@
+package manager
+
+import (
+	"context"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/constants"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/deviceplugin"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/flags"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/nsmgr"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcoptions"
+	"github.com/networkservicemesh/sdk/pkg/tools/spanhelper"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/spiffe"
+	"google.golang.org/grpc"
+)
+
+func RunNsmgr(ctx context.Context, values *flags.DefinedFlags) error {
+	span := spanhelper.FromContext(ctx, "run")
+	defer span.Finish()
+	ctx = span.Context()
+
+	var err error
+	// Capture signals to cleanup before exiting - note: this *must* be the first thing in main
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs,
```

Comment:
Signal catching should always go as early in execution as humanly possible... because if you get a signal *before* you register your notification for it... you won't receive it this way.  I'd suggest moving this either up into cmd/run.go or (if you always want to handle signals this way, into cmd.Execute().

Reply:
Yes, I've seen context with signals, will use.